WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignSectionLoader.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import WordPressKit
+
+struct SiteDesignSectionLoader {
+    static let recommendedTitle = NSLocalizedString("Best for %@", comment: "Title for a section of recommended site designs. The %@ will be replaced with the related site intent topic, such as Food or Blogging.")
+
+    static func fetchSections(vertical: SiteIntentVertical?, completion: @escaping (Result<[SiteDesignSection], Error>) -> Void) {
+        typealias TemplateGroup = SiteDesignRequest.TemplateGroup
+        let templateGroups: [TemplateGroup] = [.stable, .singlePage]
+
+        let restAPI = WordPressComRestApi.anonymousApi(
+            userAgent: WPUserAgent.wordPress(),
+            localeKey: WordPressComRestApi.LocaleKeyV2
+        )
+
+        let request = SiteDesignRequest(
+            withThumbnailSize: SiteDesignCategoryThumbnailSize.category.value,
+            withGroups: templateGroups
+        )
+
+        SiteDesignServiceRemote.fetchSiteDesigns(restAPI, request: request) { result in
+            switch result {
+            case .success(let designs):
+                let sections = assembleSections(remoteDesigns: designs, vertical: vertical)
+                completion(.success(sections))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    // Returns designs whose `group` property contains a vertical's slug
+    static func getSectionForVerticalSlug(_ vertical: SiteIntentVertical, remoteDesigns: RemoteSiteDesigns) -> SiteDesignSection? {
+        let designsForVertical = remoteDesigns.designs.filter({
+            $0.group?
+                .map { $0.lowercased() }
+                .contains(vertical.slug.lowercased()) ?? false
+        })
+
+        guard !designsForVertical.isEmpty else {
+            return nil
+        }
+
+        return SiteDesignSection(
+            designs: designsForVertical,
+            thumbnailSize: SiteDesignCategoryThumbnailSize.recommended.value,
+            categorySlug: "recommended_" + vertical.slug,
+            title: String(format: recommendedTitle, vertical.localizedTitle)
+        )
+    }
+
+    /*
```

Comment:
nit: can we use the more modern `///` for the documentation instead of the old style `/* ... */`? This way the description will serve as method documentation and will show in the summary when you option + click (same applies to all method descriptions)

Reply:
I agree! Updated in d0fac657706ef395a5cf6566591c2e8b5d6acb41.